app/lib/pantry_trak/client.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module PantryTrak
+  # Creates PantryTrak client: handling of reservartions and users
+  # For user details, see pantry_trak/users.rb
+  class Client
+    def create_user(user)
+      unless user.instance_of? PantryTrak::User
+        raise 'Requires PantryTrak::User'
+      end
+      raise user.errors.full_messages.to_sentence unless user.valid?
+
+      connection.post(create_user_path, user.to_json) do |req|
+        req.headers['Authorization'] = bearer_token
+      end.body
+    end
+
+    def create_reservation(id, user_id, event_date_id, event_slot_id)
+      payload = {
+        id: id,
+        user_id: user_id,
+        event_date_id: event_date_id,
+        event_slot_id: event_slot_id
+      }
+      connection.post(create_reservation_path, payload) do |req|
+        req.headers['Authorization'] = bearer_token
+      end.body
+    end
+
+    private
+
+    def connection
+      @connection ||= Faraday.new(base_url) do |c|
+        c.request :json
+        c.response :json
+        c.adapter Faraday.default_adapter
+      end
+    end
+
+    def bearer_token
+      payload = {
+        token: public_token,
+        time: Time.now.to_i
+      }
+      "Bearer #{JWT.encode(payload, hmac_secret, 'HS256', { typ: 'JWT' })}"
+    end
+
+    def hmac_secret
+      ENV['PANTRY_TRAK_SECRET']
+    end
+
+    def public_token
+      ENV['PANTRY_TRAK_TOKEN']
+    end
+
+    def base_url
+      ENV['PANTRY_TRAK_API_URL']
+    end
```

Comment:
Probably worth memoizing all of these since they'll be constantly accessed
```
@hmac_secret ||= ENV['PANTRY_TRAK_SECRET']
```
In a world where these are pulled at runtime from secrets manager rather than stored in the lambda env we'll be extra happy about any caching here.